packages/echo-core/src/services/analytics/analyticsModule.ts

```diff
@@ -0,0 +1,149 @@
+import { BaseError, ForbiddenError, ValidationError } from '@equinor/echo-base';
+import { IExceptionTelemetry, SeverityLevel } from '@microsoft/applicationinsights-web';
+import { EchoEnv } from '../../EchoEnv';
+import { appWithModuleName, eventNameToString } from './appInSightsLogic';
+import { AnalyticsEvent, EventName, propertyTypes } from './appInsightTypes';
+import { appInsightsInstance } from './appInsightWrapper';
+import OfflineTracker from './offlineTracker';
+
+// Based on Client Analytics Strategy
+// https://github.com/equinor/Echo/pull/14/files#diff-1e78f3b1fd220c5050d7497a73068a5e
+// Examples:
+// echo.AssetModel.Unloaded
+// echo.AssetModel.Loaded
+// ep.Document.Loaded
+// ep_xld.Documented.Opened <-- Module xld in app/service ep opened a document
+// - Object
+//   - The object that has an action performed. Such as `AssetModel` and `MeasuringTool`, they should be generic. We do **not** want both `echo.AssetModelJohanSverdrup.Loaded` and `echo.AssetModelGudrun.Loaded`. Separating which context the event was in will be a part of the **event data**.
+// - Action
+//   - The action performed on the event. This is written in past tense. (Viewed, Disabled, Enabled, Started, etc)
+
+let instCode = '';
+let userCompany = '';
+
+/**
+ * Sets the instCode logged with all events, set globally for analytics modules
+ * @param plantInstCode instCode logged with all events
+ */
+export function analyticsSetInstCode(plantInstCode: string): void {
+    instCode = plantInstCode;
+}
+
+/**
+ * Sets the company logged with all events, set globally for analytics modules
+ * @param company company logged with all events
+ */
+export function analyticsSetUserCompany(company: string): void {
+    userCompany = company.startsWith('X-') ? company.replace('X-', '') : 'Equinor';
+}
+
+/**
+ * Analytics module used to log data to application Insight
+ */
+export class AnalyticsModule {
+    moduleName: string;
+    offlineTracker: OfflineTracker;
+    constructor(moduleName: string) {
+        this.moduleName = moduleName;
+
+        const offlineThresholdSeconds = 20;
+        this.offlineTracker = new OfflineTracker(offlineThresholdSeconds, !navigator.onLine);
+        window.addEventListener('online', () => this.trackOnline);
+        window.addEventListener('offline', () => this.trackOffline);
+    }
+
+    private trackOnline(): void {
+        const evt = this.offlineTracker.setOnline();
+        if (evt) {
+            this.trackEvent2(evt.object, evt.action, {
+                minutesOffline: evt.minutesOffline,
+                actionsPerformed: evt.actionsPerformed
+            });
+        }
+    }
+
+    private trackOffline(): void {
+        this.offlineTracker.setOffline();
+    }
+
+    trackEvent2(objectName: string, actionName: string, properties: propertyTypes): void {
```

Comment:
Skjønner ikke helt navnet "trackEvent2"

Reply:
Enig, denne skal også hete trackEvent, men det er ikke lov i typescript med 2 metoder med bare forskjellig in-args. Er det noen måte å gjøre det på i typeScript?

Reply:
Hmm ikke som jeg vet om..

Reply:
It seems function overloading is possible. They need to have different signature and return types:
`````
declare function getWidget(n: number): Widget;
declare function getWidget(s: string): Widget[];
`````
Source: https://www.typescriptlang.org/docs/handbook/declaration-files/by-example.html#overloaded-functions

Not sure if this helps here: as both functions are returning with void.
Maybe just changing the name could help make the code bit more readable. Following the pattern found in `getElementById` and `getElementByName` methods:  something like `trackEventByProperties` could work?

Reply:
yes, trackEventBy is better :)